### PR TITLE
Assignment2

### PR DESCRIPTION
**Pull request by MikePHaynes**

### CODE REVIEW

**Comment by ericlplanteschool** [comment]
Will review

File: Logger/BaseLoggerMixins.cs [hunk 1]
```diff
@@ -1,7 +1,44 @@
...
+﻿using System;
+
+namespace Logger
 {
     public static class BaseLoggerMixins
     {
+        public static void Error(BaseLogger? baseLogger, string message, params object[] arguments)
+        {
+            if (baseLogger is null)
+                throw new ArgumentNullException(nameof(baseLogger));
```
Comment (ericlplanteschool) [inline Logger/BaseLoggerMixins.cs:10]: I believe this would throw an exception as baseLogger is null

File: Logger/LogFactory.cs [hunk 2]
```diff
@@ -2,10 +2,24 @@
...
+            ClassName = className;
+        }
+
+        public BaseLogger? CreateLogger()
+        {
+            if (FilePath is not null) return new FileLogger(FilePath, ClassName);
             return null;
         }
+
+        public void ConfigureFileLogger(string filePath)
```
Comment (ericlplanteschool) [inline Logger/LogFactory.cs:19]: this is never called, I believe the requirements state that it should be called from CreateLogger()

File: Logger/LogFactory.cs [hunk 3]
```diff
@@ -2,10 +2,24 @@
 {
     public class LogFactory
     {
-        public BaseLogger CreateLogger(string className)
+        private string? ClassName { get; set; }
```
Comment (ericlplanteschool) [inline Logger/LogFactory.cs:5]: I believe data members should be camel case as usually first letter capitalized is reserved for classes and methods

File: Logger/BaseLoggerMixins.cs [hunk 4]
```diff
@@ -1,7 +1,44 @@
-﻿namespace Logger
+﻿using System;
+
+namespace Logger
 {
     public static class BaseLoggerMixins
```
Comment (ericlplanteschool) [inline Logger/BaseLoggerMixins.cs:5]: this class needs unit tests

File: Logger/LogFactory.cs [hunk 5]
```diff
@@ -2,10 +2,24 @@
 {
     public class LogFactory
```
Comment (ericlplanteschool) [inline Logger/LogFactory.cs:3]: class needs unit testing

File: Logger/FileLogger.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.IO;
+
+namespace Logger
+{
+    public class FileLogger : BaseLogger
```
Comment (ericlplanteschool) [inline Logger/FileLogger.cs:6]: class needs unit testing

File: Logger/LogFactory.cs [hunk 7]
```diff
@@ -2,10 +2,24 @@
...
     {
-        public BaseLogger CreateLogger(string className)
+        private string? ClassName { get; set; }
+        private string? FilePath { get; set; }
+
+        public LogFactory(string className)
         {
-            
+            ClassName = className;
```
Comment (ericlplanteschool) [inline Logger/LogFactory.cs:10]: this should be assigned to the BaseLogger via an object initializer not passed

File: Logger/LogFactory.cs [hunk 8]
```diff
@@ -2,10 +2,24 @@
...
+        }
+
+        public BaseLogger? CreateLogger()
+        {
+            if (FilePath is not null) return new FileLogger(FilePath, ClassName);
```
Comment (ericlplanteschool) [inline Logger/LogFactory.cs:15]: does not check if the file exists

**Review by ericlplanteschool — COMMENTED** [review]
comments provided

**Comment by MikePHaynes** [comment]
@ericlplanteschool Good feedback, thank you!

File: Logger.Tests/BaseLoggerMixinsTests.cs [hunk 9]
```diff
@@ -14,7 +14,7 @@ public void Error_WithNullLogger_ThrowsException()
             // Arrange
 
             // Act
-            //BaseLoggerMixins.Error(null, "");
+            BaseLoggerMixins.Error(null, "");
```
Comment (mmwoodfo) [inline Logger.Tests/BaseLoggerMixinsTests.cs:17]: 👍🏻 This file looks good

File: Logger.Tests/FileLoggerTests.cs [hunk 10]
```diff
@@ -1,10 +1,28 @@
 ﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.IO;
 
 namespace Logger.Tests
```
Comment (mmwoodfo) [inline Logger.Tests/FileLoggerTests.cs:5]: You could probably use some more tests here to test that it does log to a new line, test logging on an existing file and a new file, etc.

File: Logger.Tests/LogFactoryTests.cs [hunk 11]
```diff
@@ -6,6 +6,24 @@ namespace Logger.Tests
     [TestClass]
     public class LogFactoryTests
```
Comment (mmwoodfo) [inline Logger.Tests/LogFactoryTests.cs:7]: 👍🏻

**Review by mmwoodfo — COMMENTED** [review]
Good job!

**Pull request closed without merging**